lib/DatabaseInfluxDB1x.js

```diff
@@ -61,19 +61,24 @@ class DatabaseInfluxDB1x extends Database{
                         retentionName = row.name;
                     }
                 });
-                callback({"name": retentionName, "time": retentionTime});
+                callback(null, {"name": retentionName, "time": retentionTime});
             }
         });
     }
 
     async applyRetentionPolicyToDB(dbname, retention, callback) {
-        this.getRetentionPolicyForDB(dbname, oldRetention => {
+        this.getRetentionPolicyForDB(dbname, (err, oldRetention) => {
+            //Check, if there was an error
+            if (err) {
+                callback(err);
+                return;
+            }
+
             //Check, if it needs to be changed, otherwise skip
             this.log.debug("old retention: " + oldRetention + " new retention: " + retention);
             if ((oldRetention.time !== null) && (oldRetention.time === retention)) {
                 this.log.debug("Retention policy for " + dbname + " remains unchanged.");
                 callback(false);
-                return;
```

Comment:
but you also need that return or? Code needs to end here, right?

Reply:
true, added it. :-)